### DISCUSSION TITLE
Flask with PyInstaller does not show images added at runtime

### DISCUSSION THREAD

**Discussion by danishansari** [pallets/flask#4180] [Q&A]
I have a simple app running on Flask, which displays images from a directory. Which works fine, until I create an executable using Pyinstaller. This executable also works when there were already some images at the time of creation, but it can't load any later added images, I guess it just bundles all the existing images. I tried with both a single file and a directory, nothing worked. I would be really grateful if somebody can give me some direction on this.

For more detail and sample code to reproduce this issue is [here](https://stackoverflow.com/questions/68153266/flask-with-pyinstaller-does-not-show-images-added-at-runtime).

**Comment by greyli** [comment]
IMO, it's more like an issue related to your own code or Pyinstaller instead of a bug in Flask (notice this issue tracker is for issues related to Flask itself), so I'm going to transfer this to our GitHub discussion.

**Reply by danishansari** [reply]
Yeah I had the same feeling, just one doubt, if it was an issue with Pyinstaller it should not be able to load image files as well, but it's reading it from directory, I can even show it using opencv/qt. It just does not show it on flask server.
If it was issue with my code, then how should not work without pyinstaller exe, but it does.

**Reply by danishansari** [reply]
one more thing if you can give an insight on, I can access any images present in static folder but not at any other location, is there any way I can access images present at any location, I think this can solve my issue.

**Comment by greyli** [comment]
> I can access any images present in static folder but not at any other location, is there any way I can access images present at any location

Flask adds a `static` route that serving the files under the static folder, so If you want to serve images at another location, you can create a custom route for it, for example:

```python
from flask import send_from_directory

@app.route('/images/<path:filename>')
def images(filename):
    base_path = path_to_the_images_folder
    return send_from_directory(base_path, filename)
```

Then you can get the image link in the template like this:

```
{{ url_for('images', filename='hello.jpg') }}
```
